libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1524,18 +1524,48 @@ bool ExpressionCompiler::visit(IndexAccess const& _indexAccess)
 				setLValueToStorageItem(_indexAccess);
 			break;
 		case DataLocation::Memory:
+			ArrayUtils(m_context).accessIndex(arrayType);
 			setLValue<MemoryItem>(_indexAccess, *_indexAccess.annotation().type, !arrayType.isByteArray());
 			break;
 		case DataLocation::CallData:
-			//@todo if we implement this, the value in calldata has to be added to the base offset
-			solUnimplementedAssert(!arrayType.baseType()->isDynamicallySized(), "Nested arrays not yet implemented.");
-			if (arrayType.baseType()->isValueType())
-				CompilerUtils(m_context).loadFromMemoryDynamic(
-					*arrayType.baseType(),
-					true,
-					!arrayType.isByteArray(),
-					false
+			if (arrayType.baseType()->isDynamicallySized())
+			{
+				unsigned int baseEncodedSize = arrayType.baseType()->category() == Type::Category::Array ?
+					dynamic_cast<ArrayType const&>(*arrayType.baseType()).calldataStride() :
+					arrayType.baseType()->calldataEncodedSize();
+				solAssert(baseEncodedSize > 0, "");
+
+				// stack layout: <base_ref> <length> <index>
+				ArrayUtils(m_context).accessIndex(arrayType, true, true);
+				// stack layout: <base_ref> <ptr_to_length>
```

Comment:
Shouldn't this be
```suggestion
				// stack layout: <base_ref> <ptr_to_tail>
```
?

The field only contains a length value if the base type is an array. Otherwise it just starts with encoding the data.

Reply:
Here we're only considering dynamically sized base types - so this should always point to a length, shouldn't it?